enhancements/installer/aws-custom-edge-machineset-local-zones.md

```diff
@@ -0,0 +1,677 @@
+
+1) Create a subnet and associate it to a route table with the internet gateway as default.
+2) Create a private subnet, associating it to a route table from the zone in the
+  parent region. The cost will be twice as the traffic should cross the zone to
+  access the internet through a Nat Gateway, which also has fees to be applied.
+3) Create a private subnet, and provide access to the node through a proxy. Same costs
+  tradeoffs of option 2.
+
+Option 1 is covered by this enhancement to create the MachineSets (phase 1)
+and to automate by the installer (phase 2).
+
+The public subnet means the node will receive a Public IP address and the Security
+Group must not expose any ports to the internet. This must be the default behavior,
+and the security group for the `worker` pool will be used on the `edge` pool.
+
+> TODO: check if the current deployment is not exposing any port to the `0.0.0.0/0` on `worker` SG
+
+#### User-workload deployments
+
+Application developers must set the tolerations for the application running for each location.
+
+The pod spec example below describes how to deploy one application on the node on the zone group `ZONE_GROUP`:
+
+```yaml
+      nodeSelector:
+        zone_group: ZONE_GROUP
+      tolerations:
+      - key: "node-role.kubernetes.io/edge"
+        operator: "Equal"
+        value: ""
+        effect: "NoSchedule"
+```
+
+<!--
+> TODO create one example of deploying the application on the same ZONE_GROUP, spreading it into different zones.
+
+> Currently only Los Angeles (LAX) location has more than one zone[1];
+
+The example below shows how to deploy one application with two replicas distributed
+between two Local Zones[1] in the same zone group[2].
+
+[1] Local Zones available in Los Angeles `us-west-2-lax-1a` and `us-west-2-lax-1b`
+[2] Local Zone group of Los Angeles `us-west-2-lax-1` -->
+
+
+#### User-workload ingress traffic
+
+To expose the applications to the internet on AWS Local Zones, application developers
+must expose their applications using AWS Application Load Balancers (ALB). The
+[ALB Operator](https://docs.openshift.com/container-platform/4.11/networking/aws_load_balancer_operator/install-aws-load-balancer-operator.html) is available through OLM on 4.11+.
+
+To explore the best of deploying applications on the edge locations, at least one new
+ALB `Ingress` must be provisioned by location to expose the services deployed on the
+zones.
+
+If the cluster-admin decides to share the ALB `Ingress` subnets between different locations,
+it will impact drastically the latency for the end-users when the traffic is routed to
+backends (compute nodes) placed in different zones that the traffic entered by Ingress.
+
+The ALB deployment is not covered by this enhancement.
+
+### Risks and Mitigations
+
+TBD
+
+### Drawbacks
+
+TBD
+
+## Design Details
+
+### Open Questions [optional]
+
+1. Is there any situation to avoid the `NoSchedule` machine pools for edge workloads?
```

Comment:
@makentenza here

Reply:
Is this asking if there's a way to avoid having the taint, or whether there's a use case for having the edge pools and worker pools run the same workloads unconditionally?

Reply:
> Is this asking if there's a way to avoid having the taint, or whether there's a use case for having the edge pools and worker pools run the same workloads unconditionally?

I mean if you'll think that using taint is the best approach for edge pools based on all scenarios and limitations described in this EP.

Reply:
IMO the taints/tolerations approach is the best and canonical way to achieve this, should we close this question?